programming-mission/pr-review.md

```diff
@@ -0,0 +1,127 @@
+
+
+## Code
+
+#### Style
+- Does it adhere to our [code style guide](https://docs.duckduckhack.com/resources/code-style-guide.html)?
+    - 4-space indents
+    - Semicolons
+    - etc.
+- Do filenames adhere to our naming conventions?
+
+#### Fatheads
+- Does the code have dependencies?
+    - If yes, make sure they're specified in a requirements.txt/gemfile/package.json/etc. file in the IA directory
+- Is [coverage data](https://docs.duckduckhack.com/programming-mission/creating-effective-fatheads.html) included?
+    - Make sure to run the Fathead coverage test to identify anything missing
+
+
+#### Spices and Goodies
+- Does it use a built-in [template](https://docs.duckduckhack.com/frontend-reference/template-groups.html)?
+    - It shouldn't use the [Base template](https://docs.duckduckhack.com/frontend-reference/template-groups.html#base-template-group) unless explicitly requested/approved by staff
+- If an API key is required, where can the DDG staff request their own copy? Make sure to ping **@moollaza** on Slack
+
+#### Perl:
+- Are all constants and private `sub`s defined outside the `handle` fn?
+- Is there a check that `$loc`, `$req`, and `$lang` are defined before using them?
+- Are all files opened and read outside of the handle function?
+- Does a CPAN module exist that we can use?
+    - Instead of implementing something yourself, check if a module already exists on CPAN
+- The Perl should not make any network requests
+    - We cannot allow the Perl to do this
+    - Any CPAN modules that make web requests are not allowed
+
+#### JavaScript:
+- Is the code wrapped in closure?
+- Is `use strict;` included at the top
+- Does this rely on any third party libraries?
+    - The developer should ask us before using external libraries. We need to add it internally if we agree to use it
+- Private variables and functions should be defined outside callback function
+- `Spice.failed()` should always be called when stopping execution prematurely
+- If the code needs to exit (e.g. the API response is missing required information) it must call `Spice.failed()`, so other Instant Answers are able to display
+- Experimental and browser-specific methods are not allowed. We need to ensure a consistent experience across all devices and browsers
+    - We support IE 9+, Edge, Chrome, Firefox, Safari, Opera, etc.
+- Will any code break if the API response is undefined?
+    - Make sure objects exist before accessing nested properties
+    - You can use `DDG.getProperty()` to get properties which may undefined (e.g. `DDG.getProperty(my_obj, 'some.property.name')`)
+
+#### CSS:
+- Is there any custom CSS?
+    - Custom CSS should be kept to a minimum. We prefer none, if possible
+- Are all CSS rules name-spaced?
+    - None of the CSS should be able to affect other Instant Answers or page elements
+- Font-colour, font-size and font-family should be defined by template variants, or our DDG classes (e.g. `.tx--14`, `.bg-clr--grey`, `.tx-clr--green`)
```

Comment:
Maybe also include a link to the [styleguide](https://duckduckgo.com/styleguide)?